doc/source/whatsnew/v1.0.0.rst

```diff
@@ -679,6 +679,7 @@ Datetimelike
 - Bug in :meth:`Series.var` failing to raise ``TypeError`` when called with ``timedelta64[ns]`` dtype (:issue:`28289`)
 - Bug in :meth:`DatetimeIndex.strftime` and :meth:`Series.dt.strftime` where ``NaT`` was converted to the string ``'NaT'`` instead of ``np.nan`` (:issue:`29578`)
 - Bug in :attr:`Timestamp.resolution` being a property instead of a class attribute (:issue:`29910`)
+- Bug in :func:`pandas.core.tools.datetimes.should_cache` failing for `deques` with ``TypeError`` when using the cache (:issue:`29403`)
```

Comment:
can you change this to ``to_datetime`` which is the user visibile function
when using ``cache=True`` (the default)

Reply:
@jreback Tests moved, whatsnew reworded. All tests have passed.